### PR TITLE
[CPU][ARM] Add fp16 capability check

### PR DESCRIPTION
**Pull request by alvoron**

### CODE REVIEW

File: src/plugins/intel_cpu/src/utils/precision_support.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@ static bool hasFP16HardwareSupport(const ov::element::Type& precision) {
 
 static bool hasBF16HardwareSupport(const ov::element::Type& precision) {
 #if defined(OPENVINO_ARCH_X86_64)
-        if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core) ||
+        if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16) ||
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/utils/precision_support.cpp:37]: That is not correct change. We support BF16 on systems with avx512_core only for debbuging/testing purposes

File: src/plugins/intel_cpu/src/plugin.cpp [hunk 2]
```diff
@@ -483,12 +484,13 @@ ov::Any Plugin::get_ro_property(const std::string& name, const ov::AnyMap& optio
         return decltype(ov::available_devices)::value_type(availableDevices);
     } else if (name == ov::device::capabilities) {
         std::vector<std::string> capabilities;
-        if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16))
+        if (hasHardwareSupport(ov::element::bf16))
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/plugin.cpp:487]: Given https://github.com/openvinotoolkit/openvino/pull/24164/files#r1574670964 I would propose to change on `dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16) || dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx2_vnni_2)`

File: src/plugins/intel_cpu/src/utils/precision_support.cpp (hunk 1 above)
Comment (alvoron) [reply src/plugins/intel_cpu/src/utils/precision_support.cpp]: reverted changes

File: src/plugins/intel_cpu/src/plugin.cpp (hunk 2 above)
Comment (alvoron) [reply src/plugins/intel_cpu/src/plugin.cpp]: fixed

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**